src/rank_llm/rerank/rank_gpt.py

```diff
@@ -151,7 +151,7 @@ def create_rank_gpt_prompt(
         query = retrieved_result["query"]
         num = len(retrieved_result["hits"][rank_start:rank_end])
 
-        max_length = 300
+        max_length = 300 * (20 / (rank_end - rank_start))
```

Comment:
let's calculate this better, maybe in a oneline function, that could be used both for LRL and rankGPT prompts?
num_words = (context size - num output tokens(current window size) )* 0.75
max_word_length = num_words / (current window size or rank_end -rank_start)

Reply:
this will likely break 20/10 reranking and out results, for now, let's do this but we can recircle?

Reply:
I completely agree, there's better ways to doing this.

Reply:
SG, I will file a bug